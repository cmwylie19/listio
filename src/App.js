import React, { useState, useEffect } from "react";
import AddTodo from "./AddTodo";
import TodoList from "./TodoList";
import DigitalClock from "./DigitalClock";

// const App = ({ db, handleToggleTodo, handleDeleteTodo, handleAddTodo }) => {
//   const [todos, setTodos] = useState([]);

// /*istanbul ignore next*/
// useEffect(() => {
//   return db
//     .table("todos")
//     .toArray()
//     .then(todos => {
//       setTodos(todos);
//     });
// });

//   return (
//     <>
//       <DigitalClock />

// <TodoList
//   todos={todos}
//   handleToggleTodo={(id, done) =>
//     handleToggleTodo(id, done, todos, setTodos)
//   }
//   handleDeleteTodo={id => handleDeleteTodo(id, todos, setTodos)}
// />

//       <AddTodo handleAddTodo={title => handleAddTodo(title, todos, setTodos)} />
//     </>
//   );
// };
// export default App;

import { makeStyles } from "@material-ui/core/styles";

import CssBaseline from "@material-ui/core/CssBaseline";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import Paper from "@material-ui/core/Paper";
import Fab from "@material-ui/core/Fab";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import ListSubheader from "@material-ui/core/ListSubheader";
import Avatar from "@material-ui/core/Avatar";
import MenuIcon from "@material-ui/icons/Menu";
import AddIcon from "@material-ui/icons/Add";
import SearchIcon from "@material-ui/icons/Search";
import MoreIcon from "@material-ui/icons/MoreVert";
import TextField from "@material-ui/core/TextField";

const useStyles = makeStyles(theme => ({
  text: {
    padding: theme.spacing(2, 2, 0),
    textAlign: "center"
  },
  paper: {
    paddingBottom: 50
  },
  list: {
    marginBottom: theme.spacing(2)
  },
  subheader: {
    backgroundColor: theme.palette.background.paper
  },
  appBar: {
    top: "auto",
    bottom: 0
  },
  toolbar: {
    display: "flex",
    flex: 1,
    width: "100%vw"
  },
  grow: {
    flexGrow: 1
  },
  fabButton: {
    position: "absolute",
    zIndex: 1,
    top: -30,
    left: 0,
    right: 0,
    margin: "0 auto"
  }
}));

export default function App({
  db,
  handleToggleTodo,
  handleDeleteTodo,
  handleAddTodo
}) {
  const classes = useStyles();
  const [todos, setTodos] = useState([]);
  const [tempTodo, setTempTodo] = useState("");
  /*istanbul ignore next*/

  useEffect(() => {
    return db
      .table("todos")
      .toArray()
      .then(todos => {
        setTodos(todos);
      });
  }, [todos]);

  return (
    <>
      <CssBaseline />
      <Paper square className={classes.paper}>
        <Typography className={classes.text} variant="h5" gutterBottom>
          <DigitalClock />
        </Typography>
        <TodoList
          todos={todos}
          handleToggleTodo={(id, done) =>
            handleToggleTodo(id, done, todos, setTodos)
          }
          handleDeleteTodo={id => handleDeleteTodo(id, todos, setTodos)}
        />
      </Paper>
      <AddTodo handleAddTodo={title => handleAddTodo(title, todos, setTodos)} />
    </>
  );
}
