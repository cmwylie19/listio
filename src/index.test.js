import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { shallow } from "enzyme";
import { indexRender } from "./index";

jest.mock("react-dom", () => ({ render: jest.fn() }));

describe("Application root", () => {
  let wrapper = shallow(<App load={true} />);
  it("should render without crashing", () => {
    const div = document.createElement("div");
    div.id = "root";
    document.body.appendChild(div);
    let spy = jest.spyOn(ReactDOM, "render");
    expect(spy).toHaveBeenCalledTimes(0);
  });
  expect(wrapper).toBeDefined();
  expect(wrapper).toMatchSnapshot();
  // expect(wrapper.find("#close")).toBeDefined();

  it("should render", () => {
    let renderSpy = jest.spyOn(ReactDOM, "render");
    indexRender();
    expect(renderSpy).toHaveBeenCalled();
  });
});
