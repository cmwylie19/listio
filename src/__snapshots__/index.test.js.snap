// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-2"
    square={true}
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-text-1"
      gutterBottom={true}
      variant="h5"
    >
      <DigitalClock />
    </WithStyles(ForwardRef(Typography))>
    <TodoList
      handleDeleteTodo={[Function]}
      handleToggleTodo={[Function]}
      todos={Array []}
    />
  </WithStyles(ForwardRef(Paper))>
  <AddTodo
    handleAddTodo={[Function]}
  />
</Fragment>
`;
